react/utils/navigation.js

```diff
@@ -0,0 +1,101 @@
+import { contains, map, path as ramdaPath, uniq, zip } from 'ramda'
+import queryString from 'query-string'
+
+const CATEGORY_TREE_MAX_DEPTH = 5
```

Comment:
CategoryTtree size is a complicated mess. I have assumed it could no more than 5, but we've find out that's not true. We will have another heuristic to compute this, which is looking for the sequence of C's in map querystring value.